compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -535,12 +535,12 @@ fn check_item_type(tcx: TyCtxt<'_>, id: hir::ItemId) {
                 return;
             };
             debug!("ItemKind::Impl {} with id {:?}", it.ident, it.owner_id);
-            if let Some(impl_trait_ref) = tcx.impl_trait_ref(it.owner_id) {
+            if let Some(impl_trait_ref) = tcx.bound_impl_trait_ref(it.owner_id.to_def_id()) {
                 check_impl_items_against_trait(
                     tcx,
                     it.span,
                     it.owner_id.def_id,
-                    impl_trait_ref,
+                    impl_trait_ref.skip_binder(),
```

Comment:
`subst_identity`. We're comparing impl items with trait items *in the context of the impl*, so the impl generic params is correct.

generally, `skip_binder` is pretty much always wrong unless we 1. rebind the binder later on, or 2. only look at something which isn't a `Ty` or the outermost ty constructor or something